### ISSUE TITLE
What is different between task.build_model() and factory.build_*_model ?

### ISSUE THREAD

**Issue by DoHoNi** [tensorflow/models#10557]
I am training my model using codes in models/official/vision/beta.
I am curious about the difference between taking the **registered task** and **using the factory** to return a specific model.
I recognized that a way getting a model is calling registered tasks and calling build_model() when I start to train my model.
In the following way,
https://github.com/tensorflow/models/blob/871c4e0a393ef4385534bee55354a5df8aa1ccf4/official/vision/beta/train.py#L56
https://github.com/tensorflow/models/blob/871c4e0a393ef4385534bee55354a5df8aa1ccf4/official/core/train_utils.py#L221

However, I found out the way is different when I export my model.
I used the code ( https://github.com/tensorflow/models/blob/master/official/vision/beta/serving/export_saved_model.py) to export my model.
In this case, they use 'factory' defined models/official/vision/beta/modeling/factory (https://github.com/tensorflow/models/blob/master/official/vision/beta/modeling/factory.py) to get a model.  
https://github.com/tensorflow/models/blob/c1c9bb0fb019de246ea268b1f15e0c76a76138f1/official/vision/beta/serving/image_classification.py#L31-L38

So I was very confused, and I wonder why you make and use 'factory' in models/official/vision/beta/modeling instead of calling a model using a task.

**Comment by sushreebarsa** [comment]
@DoHoNi  Generally task.build_model() is used for loading/iterating over Datasets, initializing the Model/Criterion and calculating the loss. Could you provide the working URLs for standalone code as the given code is not working properly.
Thank you!
